### PR TITLE
Tweak BSoD to support semver labels

### PR DESCRIPTION
**Pull request by jchannon**
The build script does not support us putting out semver pre-releases (like `-alpha` and so on) because in order to that do that, we need to set the `AssemblyInformalVersionAttibute` to a different string than the `AssemblyVersionAttribute`. The latter only supports `major.minor.revision` versions, while the former is the one that should include the pre-release suffix.

This pull-requests adds a second parameter to the tasks, letting us specify an optional `label`. If the label is present, then `AssemblyInformalVersionAttribute` will be a combination of `version-lable` otherwise it will have the same value as `AssemblyVersionAttribute`
## Todo
- [x] Update description with information about `label`
- [x] Update `prep_release` task to support `label`
- [x] Update `tag_nancy` task to support `label`
- [x] Update `update_project` task to support `label`
- [x] Update `update_version` task to support `label`

### CODE REVIEW

File: rakefile.rb [hunk 1]
```diff
@@ -3,20 +3,20 @@
...
-require 'FileUtils'
+require 'fileutils'
 
 task :default do
   puts "Nancy Release Script"
   puts
   puts "** Usage **"
   puts
-  puts "nancy:prep_release[version]"
+  puts "nancy:prep_release[version,label]"
```
Comment (thecodejunkie) [inline rakefile.rb:13]: I think we need to make it clear that the label is optional

File: rakefile.rb [hunk 2]
```diff
@@ -235,10 +242,12 @@ def self.logged_chdir(dir, &block)
   end
 
   desc "Updates #{SHARED_ASSEMBLY_INFO} version"
-  customassemblyinfo :update_version, :version do |asm, args|
+  customassemblyinfo :update_version, [:version, :label] do |asm, args|
       asm.input_file = SHARED_ASSEMBLY_INFO
       asm.version = args.version if !args.version.nil?
-      asm.assembly_informational_version = args.version if !args.version.nil?
+      aiv = args.version if !args.version.nil?
```
Comment (khellang) [inline rakefile.rb]: Can we call this `informational_version`?

File: rakefile.rb [hunk 3]
```diff
@@ -124,36 +124,43 @@ def self.logged_chdir(dir, &block)
...
+      Git.tag "v#{versionInfo}", false, "Tagged v#{versionInfo}"
       Git.push 'origin master'
       Git.push_tags
     end
   end
 
-  task :update_project, :project, :version do |task, args|
-    puts "Updating: #{args.project} to v#{args.version}"
+  task :update_project, [:project, :version, :label] do |task, args|
+    versionInfo = args.version if !args.version.nil?
```
Comment (khellang) [inline rakefile.rb]: `version_info`?

File: rakefile.rb (hunk 1 above)
Comment (jchannon) [inline rakefile.rb:13]: Added below

**Pull request merged**